src/coreclr/src/jit/morph.cpp

```diff
@@ -13912,19 +13953,38 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 // normalization.
                 if (temp->OperIs(GT_LCL_VAR))
                 {
-#ifdef DEBUG
-                    // We clear this flag on `temp` because `fgMorphLocalVar` may assert that this bit is clear
-                    // and the node in question must have this bit set (as it has already been morphed).
-                    temp->gtDebugFlags &= ~GTF_DEBUG_NODE_MORPHED;
-#endif // DEBUG
-                    const bool forceRemorph = true;
-                    temp                    = fgMorphLocalVar(temp, forceRemorph);
-#ifdef DEBUG
-                    // We then set this flag on `temp` because `fgMorhpLocalVar` may not set it itself, and the
-                    // caller of `fgMorphSmpOp` may assert that this flag is set on `temp` once this function
-                    // returns.
-                    temp->gtDebugFlags |= GTF_DEBUG_NODE_MORPHED;
-#endif // DEBUG
+                    unsigned  lclNum  = temp->AsLclVarCommon()->GetLclNum();
```

Comment:
looks like it is expected to go into this block when `addCastToType` is set to non-void but it won't for `GT_LCL_FLD`.

Reply:
I added an assert that `addCastToType` is `TYP_VOID` when we don't have a GT_LCL_VAR